frontend/src/containers/Subscription/SubscriptionFeed.js

```diff
@@ -1,16 +1,55 @@
 import React, { Component } from "react";
 import { connect } from "react-redux";
 import PropTypes from "prop-types";
+import { Button } from "react-bootstrap";
 
 import { PaperCard, CollectionCard, ReviewCard } from "../../components";
 import { authActions } from "../../store/actions";
 
 import "./SubscriptionFeed.css";
 
 class SubscriptionFeed extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            subscriptions: [],
+            recommendations: [],
+            finished: false,
+            start: 0,
+            recoNum: 10,
+        };
+        this.paperCardMaker = this.paperCardMaker.bind(this);
+        this.reviewCardMaker = this.reviewCardMaker.bind(this);
+        this.collectionCardMaker = this.collectionCardMaker.bind(this);
+        this.clickMoreButton = this.clickMoreButton.bind(this);
+        this.addRecoToSub = this.addRecoToSub.bind(this);
+    }
+
     componentDidMount() {
         this.props.onGetSubscriptions()
-            .catch(() => {});
+            .then(() => {
+                this.setState({
+                    subscriptions: this.props.subscriptionItems,
+                });
+                this.props.onGetRecommendations()
+                    .then(() => {
+                        this.setState({
+                            recommendations: this.props.recommendationItems,
+                        });
+                        this.addRecoToSub();
+                        if (this.props.subscriptionFinished) {
+                            if (this.state.recommendations.length <= 0) {
+                                this.setState({
+                                    finished: true,
+                                });
+                            } else {
+                                this.setState({
+                                    recoNum: 20,
```

Comment:
[Not important] I think 'count' is more consistent with other codes.

Reply:
Oh good